ambry-api/src/main/java/com.github.ambry/rest/RequestPath.java

```diff
@@ -91,9 +95,18 @@ public static RequestPath parse(String path, Map<String, Object> args, List<Stri
     // test if the last segment is a sub-resource
     RestUtils.SubResource subResource = null;
     int lastSlashOffset = path.lastIndexOf('/');
+    long blobSegmentNum = NO_BLOB_SEGMENT_IDX_SPECIFIED;
+    System.out.println(Arrays.asList(path.split("/")));
     if (lastSlashOffset > offset) {
       try {
-        subResource = RestUtils.SubResource.valueOf(path.substring(lastSlashOffset + 1));
+        String[] fields = path.split("/");
+        if (RestUtils.SubResource.Segment.toString().equals(fields[fields.length - 2])) {
+          subResource = RestUtils.SubResource.valueOf(fields[fields.length - 2]);
+          blobSegmentNum = Long.valueOf(fields[fields.length - 1]);
```

Comment:
Don't we need to handle NumberFormatException?  What if user types Segment/foo?

Reply:
will throw BadRequest now with non-integer input